mimium-lang/src/utils/half_float.rs

```diff
@@ -0,0 +1,51 @@
+use half::f16;
+use std::fmt::Display;
+/// Half-Precision floating point type that can be converted from 64bit float with truncation checking.
+pub const ALLOWED_ERROR: f64 = 0.00001;
+
+#[derive(Clone, Copy, Debug, PartialEq, PartialOrd)]
+pub struct HFloat(f16);
+
+impl Display for HFloat {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl From<f16> for HFloat {
+    fn from(value: f16) -> Self {
+        Self(value)
+    }
+}
+impl Into<f64> for HFloat {
+    fn into(self) -> f64 {
+        self.0.to_f64()
+    }
+}
+
+impl TryFrom<f64> for HFloat {
+    type Error = ();
+
+    fn try_from(value: f64) -> Result<Self, Self::Error> {
+        let hv = f16::from_f64(value);
+        let error = (hv.to_f64() - value).abs();
+        if error < ALLOWED_ERROR {
+            Ok(Self(f16::from_f64(value)))
+        } else {
+            Err(())
+        }
+    }
+}
```

Comment:
Now, compiler tries lossy conversion at first and compares its value to the original.
With this way, most of integer literals will be loaded through MoveImmF.
Currently, allowed error is global constant `0.00001` but maybe it should be controllable value by user.